ecommerce/api.py

```diff
@@ -182,6 +184,9 @@ def apply_user_discounts(request):
         if not check_discount_for_products(discount, basket):
             return
 
+        if not discount.check_validity(user):
+            return
+
```

Comment:
nit picking, but can the two `if` statements between line 184 and 189 be combined with an OR and just `return`?